### DISCUSSION TITLE
What does grid control switch do?

### DISCUSSION THREAD

**Discussion by whulshof** [WillCodeForCats/solaredge-modbus-multi#384] [Q&A]
I'm using this integration for more than a year now. Just instead of the app, as sensor. I only wanted to have a switch to turn production off when the prices get negative. Now I noticed the power control options in the latest release, although I think the grid control switch was already present before. 
I'm a bit reluctant to "just" try the switch. 
Can anybody tell me if it is simply the on/off switch of the inverter?
It would be great news, because I then can automate switching off the solar production when I would have to pay for returned energy. I rather use as much as possible from the grid in that situation, ofc.

**Comment by WillCodeForCats** [comment]
No it is not an on/off switch. I don't actually know what it does. Mainly I added it because it's mentioned in the docs.

Production control is an existing function as part of site limit controls. To limit production set `select.solaredge_i1_limit_control_mode` to `Production Control` and then set `number.solaredge_i1_site_limit` to 0.

**Reply by whulshof** [reply]
So with these parameters I could make a template switch and use that for my switching. Great!

**Reply by whulshof** [reply]
Completely different matter: the latest release of your integration seems to have a lot of "battery" stuff. I would like to get into that more, but I haven't got the faintest where to get info for Solaredge --> battery solutions, apart from the fact that their products are named StorEdge. Could you give me any direction where to look?

**Reply by WillCodeForCats** [reply]
I don't have batteries on my system so unfortunately I can't really help with them. Hopefully someone else will be able answer battery-related questions, or there are other discussions that have battery-related stuff in them.

**Reply by deqw** [reply]
@whulshof Maybe take a look here: [https://www.solaredge.com/en/products/residential/storage-and-backup/solaredge-home-battery-48v](url)

Please note that a lot of SolarEdge information, pictures, videos on this subject show options and software that does not exist in real life, I found out.... It's all alpha and/or beta stadium. So your only option for now is to fiddle with modbus and HA integrations if you want to do advanced control. Yesterday I noted they have a beta program you can enrol into, so I gave that a try and now waiting for an answer. Battery is working very fine BTW.

**Reply by deqw** [reply]
For the Grid Control: "SolarEdge inverters can connect to an external device, which can control active and reactive power according to commands sent by the grid operator". Please refer to: [https://knowledge-center.solaredge.com/sites/kc/files/application_note_power_control_configuration.pdf](url)

**Comment by jovana** [comment]
> I'm using this integration for more than a year now. Just instead of the app, as sensor. I only wanted to have a switch to turn production off when the prices get negative. Now I noticed the power control options in the latest release, although I think the grid control switch was already present before.
> I'm a bit reluctant to "just" try the switch.
> Can anybody tell me if it is simply the on/off switch of the inverter?
> It would be great news, because I then can automate switching off the solar production when I would have to pay for returned energy. I rather use as much as possible from the grid in that situation, ofc.

Probably this can help you,  https://github.com/WillCodeForCats/solaredge-modbus-multi/discussions/296 I need to have the inverter also switched off when pricing is negative.
